### PR TITLE
cubical-mini: init at nightly-20241214

### PR DESCRIPTION
**Pull request by thelissimus**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (steeleduncan) [inline pkgs/development/libraries/agda/cubical-mini/default.nix:11]: Is it possible to pin this to a stable version? e.g. v0.5?

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (steeleduncan) [inline pkgs/development/libraries/agda/cubical-mini/default.nix:26]: `runHook preBuild` before and `runHook postBuild` after

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (steeleduncan) [inline pkgs/development/libraries/agda/cubical-mini/default.nix:33]: with lib is now deprecated #208242

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (steeleduncan) [inline pkgs/development/libraries/agda/cubical-mini/default.nix:30]: I'm not familiar with this package, so it might not be for the best, but generally if a makefile exists the derivation will call make automatically

Potentially you can setup the cabal folders in `preBuild` and leave the `buildPhase` out

**Review by steeleduncan — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (thelissimus) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:11]: Yes, but then it will only build on Linux, not MacOS. See: https://github.com/cmcmA20/cubical-mini/pull/63

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (thelissimus) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:30]: > Potentially you can setup the cabal folders in preBuild and leave the `buildPhase` out

I tried building it locally omitting the `buildPhase` and it did not work. Turns out, one, in fact, needs to override `buildPhase` to call `make`: [`agda.section.md`](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/agda.section.md#building-agda-packages-building-agda-packages)

While writing the derivation I was guided by the existing [`cubical`](https://github.com/NixOS/nixpkgs/blob/21d7df2cae8e141691e863aa54b75ca82c946afd/pkgs/development/libraries/agda/cubical/default.nix#L19-L26) package, because both libraries are built similarly.

Taking into account your suggestions regarding build phases, I ended up with this for now:

```nix
preBuild = ''
  export HOME=$TMP
  mkdir $HOME/.cabal
  touch $HOME/.cabal/config
'';

buildPhase = ''
  runHook preBuild
  make
  runHook postBuild
'';
```

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (thelissimus) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:26]: Done.

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (thelissimus) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:11]: @steeleduncan So, can I leave it at nightly?

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (thelissimus) [reply pkgs/development/libraries/agda/cubical-mini/default.nix]: Done.

**Review by steeleduncan — COMMENTED** [review]
I don't have commit access, and don't maintain a similar package, so it doesn't make sense for me to leave an approving review. However I wouldn't want to block the PR in any way

My suggestion now would be to find someone who has merged a similar package into `nixpkgs` recently, and request a review from them

**Comment by thelissimus** [comment]
@ncfavier I noticed that you recently merged a PR for agda-stdlib (https://github.com/NixOS/nixpkgs/pull/339651). Would you mind taking a look at this PR and merge it if everything looks good?

File: pkgs/development/libraries/agda/cubical-mini/default.nix
Comment (ncfavier) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:26]: The point of `preBuild` is generally to allow users to run commands before the `buildPhase`. If we use it ourselves, then if a user tries to override it the package will not build any more. We should just move these commands to `buildPhase`, between `runHook preBuild` and `make.

Comment (ncfavier) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:26]: Also maybe add a comment explaining why this dance is needed (the Makefile uses `cabal run ...`, and the default Cabal config makes it require an Internet connection).

Comment (thelissimus) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:27]: Changed and pushed.

Comment (ncfavier) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:27]: > Also maybe add a comment explaining why this dance is needed (the Makefile uses `cabal run ...`, and the default Cabal config makes it require an Internet connection).

Comment (thelissimus) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:27]: Sorry I didn't see your second comment the first time, because I didn't refresh the GitHub page. One moment.

Comment (thelissimus) [reply pkgs/development/libraries/agda/cubical-mini/default.nix:29]: Added the comment too.

**Review by ncfavier — APPROVED** [review]
Thanks!

**Pull request merged**
